l10n_it_intrastat_statement/models/intrastat.py

```diff
@@ -0,0 +1,2214 @@
+                rcd += line._prepare_export_line()
+                file_content += rcd
+            # Section 4
+            for line in self.sale_section4_ids:
+                rcd = self._prepare_export_prefix("sale")
+                rcd += '{:1s}'.format("4")
+                rcd += '{:5s}'.format(str(line.sequence).zfill(5))
+                rcd += line._prepare_export_line()
+                file_content += rcd
+
+        # Validtation data
+        if not file_content:
+            raise ValidationError(
+                _('Nothing to export'))
+        if not self.sale_section1_ids \
+                and not self.sale_section2_ids \
+                and not self.sale_section3_ids \
+                and not self.sale_section4_ids \
+                and not self.purchase_section1_ids \
+                and not self.purchase_section2_ids \
+                and not self.purchase_section3_ids \
+                and not self.purchase_section4_ids:
+            raise ValidationError(
+                _('Statement without lines'))
+
+        return file_content
+
+    @api.one
+    def compute_statement(self):
+        # Unlink existing lines
+        self._unlink_sections()
+        # Setting period
+        period_date_start = None
+        period_date_stop = None
+        day = self.env.user.company_id.fiscalyear_last_day
+        month = self.env.user.company_id.fiscalyear_last_month
+        date_start_year = date(self.fiscalyear, month, day)
+        if self.period_type == 'M':
+            period_date_start = datetime(date_start_year.year,
+                                         self.period_number,
+                                         1)
+            # Last date of month
+            if not self.period_number == 12:
+                period_date_work = datetime(date_start_year.year,
+                                            self.period_number + 1,
+                                            1)
+                period_date_stop = period_date_work - timedelta(days=1)
+            else:
+                period_date_stop = datetime(date_start_year.year, 12, 31)
+
+        else:
+            if self.period_number == 1:
+                period_date_start = datetime(date_start_year.year, 1, 1)
+                period_date_stop = datetime(date_start_year.year, 3, 31)
+            elif self.period_number == 2:
+                period_date_start = datetime(date_start_year.year, 4, 1)
+                period_date_stop = datetime(date_start_year.year, 6, 30)
+            elif self.period_number == 3:
+                period_date_start = datetime(date_start_year.year, 7, 1)
+                period_date_stop = datetime(date_start_year.year, 9, 30)
+            elif self.period_number == 4:
+                period_date_start = datetime(date_start_year.year, 10, 1)
+                period_date_stop = datetime(date_start_year.year, 12, 31)
+
+        # Search intrastat lines
+        domain = [('move_id.date', '>=', period_date_start),
```

Comment:
Nel calcolo di questi filtri per data, non vengono per nulla presi in considerazione i campi `date_start` e `date_stop`.
Ha senso farli impostare all'utente se poi non vengono considerati?

Reply:
Gestito l'errore sul Trimestre e nascosto i campi date_start e date_end

Reply:
@glauco70 vedi https://github.com/linkitspa/l10n-italy/pull/7 per la gestione del periodo

Reply:
@SimoRubi grazie della PR.. portiamo le stesse correzioni/migliorie anche sulla 11